flash/tabular/classification/model.py

```diff
@@ -59,82 +61,72 @@ class TabularClassifier(ClassificationTask):
     """
 
     required_extras: str = "tabular"
+    backbones: FlashRegistry = FlashRegistry("backbones") + PYTORCH_TABULAR_BACKBONES
 
     def __init__(
         self,
-        num_features: int,
-        num_classes: int,
-        embedding_sizes: Optional[List[Tuple[int, int]]] = None,
+        embedding_sizes: list,
+        categorical_fields: list,
+        cat_dims: list,
+        num_categorical_fields: int,
+        num_numerical_fields: int,
+        output_dim: int,
```

Comment:
It seems like we could remove some arguments here. Given `num_features` and `embedding_sizes` you have:
```
num_categorical_fields = len(embedding_sizes)
num_numerical_fields = num_features - num_categorical_fields
```
Or at least `num_categorical_fields = len(categorical_fields)`